controller/src/test/java/io/pravega/controller/store/stream/PravegaTablesInt96CounterTest.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.pravega.controller.store.stream;
+
+
+import io.pravega.common.concurrent.ExecutorServiceHelpers;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.lang.Int96;
+import io.pravega.controller.mocks.SegmentHelperMock;
+import io.pravega.controller.server.SegmentHelper;
+import io.pravega.controller.server.security.auth.GrpcAuthHelper;
+import io.pravega.controller.store.PravegaTablesStoreHelper;
+import io.pravega.controller.store.TestOperationContext;
+import io.pravega.controller.store.Version;
+import io.pravega.controller.store.VersionedMetadata;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+
+import static io.pravega.shared.NameUtils.TRANSACTION_ID_COUNTER_TABLE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.*;
+
+public class PravegaTablesInt96CounterTest {
```

Comment:
Ideally, if we have 2 implementations for `Int96Counter`, both should be passing the same "common" test cases, by extending an "abstract" test class with the common test methods (and you can add implementation specific tests in each of the test classes, plus the specific setup logic). This would help to ensure that the different implementations keep the same behavior, which may be important in upgrade scenarios.

Reply:
I have updated the test case as suggested.